lib/lambda/sinkMainProcessors.ts

```diff
@@ -222,86 +268,79 @@ export const insertNewSeatoolRecordsFromKafkaIntoMako = async (
   topicPartition: string,
 ) => {
   const makoDocTimestamps = await getMakoDocTimestamps(kafkaRecords);
+  const seatoolRecordsForMako: { id: string; [key: string]: unknown }[] = [];
 
-  const seatoolRecordsForMako = kafkaRecords.reduce<{ id: string; [key: string]: unknown }[]>(
-    (collection, kafkaRecord) => {
-      try {
-        const { key, value } = kafkaRecord;
+  for (const kafkaRecord of kafkaRecords) {
```

Comment:
This had to get changed because of the asynchronous nature of the comparison of comparing  what is coming in from seatool and what we have in onemac .  Functionally its all the same though.

Reply:
You could've just wrapped it in `Promise.all`, no? Like in `sinkMain.ts`

Reply:
I'd like you to try this:

```ts
const seatoolRecordsForMako = await kafkaRecords.reduce(
  async (collection, kafkaRecord) => {
   // code
},
  Promise.resolve([] as { id: string; [key: string]: unknown }[])
);
```